### DISCUSSION TITLE
Валидация ресурсов (в том числе с использованием терминологического сервиса)

### DISCUSSION THREAD

**Discussion by Vrodvi** [fhir-ru/core#93] [General]
- [1] Каким образом можно валидировать ресурсы на соответствие профилям, определённым в руководстве по реализации FHIR-ru Core?

  -- Будет ли реализован встроенный в семантическую вики инструмент для валидации или будет организована возможность экспорта профилей в стандартный формат (StructureDefinition) для использования в сторонних валидаторах?

- [1.1] В том числе каким образом можно валидировать кодированные значения на соответствие определённым в руководстве наборам значений? 

  -- Будет ли сообщество FHIR-ru разворачивать терминологический сервис FHIR (как определено тут - https://www.hl7.org/fhir/terminology-service.html) для ведения и работы с терминологическими ресурсами или валидация терминологий предполагается каким-то альтернативным образом?

  -- К примеру, если в профиле Condition код заболевания будет определён как код из набора значений (valueSet) "все коды из справочника ФНСИ - "МКБ-10" (https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.11.1005)", то каким образом мы сможем валидировать экземпляры ресурсов этого профиля на соответствие указанному справочнику (проблема в том что справочники ФНСИ не являются FHIR-ресурсами codeSystem)?

**Comment by eugenekogan-spb** [comment]
@niquola если есть альтернативные решения, их наверное стоит изложить тут.
Из обсуждения на встрече 22.03.22 http://fhir-ru.zendoc.me/meetings.2022-03-22 :
- примером справочника является МКБ-10 (16000 позиций), SNOMED не требуется
- часть терминологии лежит на НСИ ЕГИСЗ и будет меняться, надо подписываться, есть API

**Comment by eugenekogan-spb** [comment]
Из обсуждения на встрече 2022-03-29 http://fhir-ru.zendoc.me/meetings.2022-03-29
Решили пока не создавать сервер терминологий, поскольку его развертывание и поддержание трудоемко.
Валидацию @niquola сделает в zen, ведение терминологии будет возможно здесь же, в объеме, необходимом для валидации отдельных экземпляров ресурсов. Примеры будут взяты из https://github.com/fhir-ru/zendoc/issues/16 .

**Reply by Vrodvi** [reply]
На прошедшем недавно митапе (https://youtu.be/1WalljTHuus) Grahame Grieve (FHIR product director for HL7 International) обосновал важность использования терминологического сервера и настоятельно рекомендовал использование такого сервера для работы с терминологиями.

В связи с этим огорчает быстрота, с которой было отброшено предложение об использовании терминологического сервера.
Имеется опасение, что предложение Николая будет работать только для небольших справочников, для которых наборы значений можно составить перечислением кодов, а для более сложных случаев (наборы значений определённые по правилам включение\исключения) мы столкнёмся с невозможностью осуществлять валидацию кодированных значений.
Надеюсь, что опасения будут напрасны, поэтому ждём реализации предложенного Николаем подхода.

**Comment by eugenekogan-spb** [comment]
@Vrodvi возможно, дискуссию стоит пометить как завершенную.

**Reply by Vrodvi** [reply]
Закрывать дискуссию можно будет только после получения конкретных результатов - т.е. описанного и работающего алгоритма проверки экземпляров ресурсов на соответствие профилям (в том числе с проверкой кодированных значений).

**Reply by JustAsWalker** [reply]
Соглашусь с @Vrodvi, по крайней мере до тех пор, пока мы не договорились о конкретных целях использования дискуссий (discussions) и задач (issue). На примере данного вопроса - если обсуждение закончено, и мы пришли к какому-то решению, то дискуссию теоретически можно закрыть, но закрыть её с комментарием о том, что на основании неё создана задача (или задачи) на доработку zendoc (или если уже все доработано, на подготовку примеров, как использовать для валидации). Тогда дальнейшую проверку, о которой упомянул Виталий, можно будет осуществлять уже в рамках задач.

**Reply by eugenekogan-spb** [reply]
Да, я тоже соглашусь, так будет четче.
Созданы Issue
https://github.com/fhir-ru/core/issues/107 - валидация
https://github.com/fhir-ru/core/issues/108 - ведение справочников
